meteor/client/ui/Shelf/PieceCountdownPanel.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react'
+import * as _ from 'underscore'
+import ClassNames from 'classnames'
+import {
+	RundownLayoutBase,
+	RundownLayoutPieceCountdown,
+	DashboardLayoutPartCountdown,
+} from '../../../lib/collections/RundownLayouts'
+import { RundownLayoutsAPI } from '../../../lib/api/rundownLayouts'
+import { dashboardElementPosition, getUnfinishedPieceInstancesReactive } from './DashboardPanel'
+import { withTracker } from '../../lib/ReactMeteorData/ReactMeteorData'
+import { MeteorReactComponent } from '../../lib/MeteorReactComponent'
+import { RundownUtils } from '../../lib/rundown'
+import { RundownTiming, TimingEvent } from '../RundownView/RundownTiming/RundownTiming'
+import { RundownPlaylist } from '../../../lib/collections/RundownPlaylists'
+import { PieceInstance } from '../../../lib/collections/PieceInstances'
+import { VTContent } from '@sofie-automation/blueprints-integration'
+interface IPieceCountdownPanelProps {
+	visible?: boolean
+	layout: RundownLayoutBase
+	panel: RundownLayoutPieceCountdown
+	playlist: RundownPlaylist
+}
+
+interface IPieceCountdownPanelTrackedProps {
+	livePieceInstance?: PieceInstance
+}
+
+interface IState {
+	displayTimecode: number
+}
+
+export class PieceCountdownPanelInner extends MeteorReactComponent<
+	IPieceCountdownPanelProps & IPieceCountdownPanelTrackedProps,
+	IState
+> {
+	constructor(props) {
+		super(props)
+		this.state = {
+			displayTimecode: 0,
+		}
+		this.updateTimecode = this.updateTimecode.bind(this)
+	}
+
+	componentDidMount() {
+		window.addEventListener(RundownTiming.Events.timeupdate, this.updateTimecode)
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener(RundownTiming.Events.timeupdate, this.updateTimecode)
+	}
+
+	updateTimecode(e: TimingEvent) {
+		let timecode = 0
+		if (this.props.livePieceInstance && this.props.livePieceInstance.startedPlayback) {
+			const vtContent = this.props.livePieceInstance.piece.content as VTContent | undefined
+			const sourceDuration = vtContent?.sourceDuration || 0
+			const startedPlayback = this.props.livePieceInstance.startedPlayback
+			if (startedPlayback && sourceDuration > 0) {
+				timecode = e.detail.currentTime - (startedPlayback + sourceDuration)
```

Comment:
This also needs to take into account `vtContent?.seek` for seeked clips.

Reply:
Done